com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/ReportGenerator.java

```diff
@@ -126,60 +125,52 @@ private void parseFromDisk() throws IOException {
         }
     }
 
-    /**
-     * Gathers file sizes (both compressed and uncompressed) and encryption flags
-     * from file entries stored in a game.darc file.
-     */
-    private void parseFromDarc() throws IOException {
-
-        String rootDir = FilenameUtils.concat(project.getRootDirectory(), project.getBuildDirectory());
-        String archiveIndex = FilenameUtils.concat(rootDir, "game.arci");
-        String archiveData = FilenameUtils.concat(rootDir, "game.arcd");
-        String manifest = FilenameUtils.concat(rootDir, "game.dmanifest");
-
-        ArchiveReader ar = new ArchiveReader(archiveIndex, archiveData, manifest);
-
-        ar.read();
-
-        List<ArchiveEntry> archiveEntries = ar.getEntries();
-        for (int i = 0; i < archiveEntries.size(); i++) {
-            ArchiveEntry archiveEntry = archiveEntries.get(i);
-            long compressedSize = archiveEntry.getCompressedSize() != -1 ? archiveEntry.getCompressedSize() : archiveEntry.getSize();
+    private void parseFromPublisher(Publisher p) {
+        Map<String, ArchiveEntry> entries = p.getEntries();
+        for (String fileName : entries.keySet()) {
+            ArchiveEntry archiveEntry = entries.get(fileName);
+            long compressedSize = archiveEntry.isCompressed() ? archiveEntry.getCompressedSize() : archiveEntry.getSize();
             boolean encrypted = archiveEntry.isEncrypted();
+            boolean excluded = true;
 
-            if (this.resources.containsKey(archiveEntry.getFilename())) {
-                ResourceEntry resEntry = this.resources.get(archiveEntry.getFilename());
-                resEntry.compressedSize = compressedSize;
-                resEntry.size = archiveEntry.getSize();
-                resEntry.encrypted = encrypted;
-            } else {
-                ResourceEntry resEntry = new ResourceEntry(archiveEntry.getFilename(),
+            ResourceEntry resEntry = new ResourceEntry(archiveEntry.getRelativeFilename(),
                         archiveEntry.getSize(),
                         compressedSize,
-                        encrypted);
-
-                this.resources.put(archiveEntry.getFilename(), resEntry);
-            }
+                        encrypted,
+                        excluded);
 
+            this.excludedResources.put(archiveEntry.getRelativeFilename(), resEntry);
         }
-
-        ar.close();
     }
 
+    private void parseArchiveBuilder(ArchiveBuilder archiveBuilder) {
```

Comment:
If we archive bundle - we already have all needed data there. it doesn't make sense to read it from darc archive